src/test/java/pro/trevor/tankgame/util/DummyState.java

```diff
@@ -0,0 +1,12 @@
+package pro.trevor.tankgame.util;
+
+import pro.trevor.tankgame.state.State;
+
+public class DummyState extends State {
+
+    public static final State DUMMY_STATE = new DummyState();
```

Comment:
This being static makes our unit tests susceptible to test case ordering and stale data from prior test cases. I think all instances of DUMMY_STATE need to be replaced with `new DummyState()` so that unit test cases are completely independent of one another.

Reply:
ACK